nifi-extension-bundles/nifi-extension-utils/nifi-file-transfer/src/main/java/org/apache/nifi/processor/util/file/transfer/FetchFileTransfer.java

```diff
@@ -363,7 +383,39 @@ private void performCompletionStrategy(final FileTransfer transfer, final Proces
                     transfer.ensureDirectoryExists(flowFile, new File(absoluteTargetDirPath));
                 }
 
-                final String destinationPath = String.format("%s/%s", absoluteTargetDirPath, simpleFilename);
+                String destinationFileName = simpleFilename;
+                final FileInfo remoteFileInfo = transfer.getRemoteFileInfo(flowFile, absoluteTargetDirPath, destinationFileName);
+                if (remoteFileInfo != null) {
+                    final String strategy = context.getProperty(MOVE_CONFLICT_RESOLUTION).getValue();
+                    switch (strategy.toUpperCase()) {
+                        case FileTransfer.CONFLICT_RESOLUTION_REPLACE:
+                            try {
+                                transfer.deleteFile(flowFile, absoluteTargetDirPath, destinationFileName);
+                            } catch (final IOException deleteEx) {
+                                getLogger().warn("Failed to delete existing destination file {} on {}:{}. Move will be attempted regardless.",
```

Comment:
I don't think it is necessary to mention the next action, the message can be as follows:

```suggestion
                                getLogger().warn("Failed to delete existing destination file {} on {}:{}",
```